velox/exec/tests/RowContainerTest.cpp

```diff
@@ -277,8 +278,50 @@ class RowContainerTest : public exec::test::RowContainerTestBase {
     return rowContainer;
   }
 
+  /// Helper function for the floating point comparison tests.
+  /// Reorders the initialOrder vector depending on the
+  /// requested ordering options (ascending vs descending)
+  /// and whether or not nulls are first.
+  /// The inital order vector is ordered in ascending order of
+  /// vector elements and, if needed, reordered.
   template <typename T>
-  void testCompareFloats(TypePtr type, bool ascending, bool nullsFirst) {
+  void setupFloatTestExpectedOutput(
```

Comment:
Looking at this method some more, it seems like it takes a list of values sorted in ascending order with nulls first and re-orders the list according to specified ascending and nullsFirst flags. If that's the case, I suggest to rename this method to countLeadingNulls and implement it like this:

```
  // Count number of nulls at the start of the list.
  template <typename T>
  static size_t countLeadingNulls(std::vector<std::optional<T>>& values) {
    int32_t numNulls = 0;
    for (auto i = 0; i < values.size(); ++i) {
      if (!values[i].has_value()) {
        ++numNulls;
      } else {
        break;
      }
    }

    return numNulls;
  }

  // Given a list of values sorted in ascending order with nulls first re-orders
  // the elements to reverse the order and / or pu nulls last.
  template <typename T>
  void changeSortingOrder(
      std::vector<std::optional<T>>& values,
      bool ascending,
      bool nullsFirst) {
    if (ascending) {
      if (nullsFirst) {
        // Nothing to do. 'values' are expected to be already sorted in
        // ascending order with nulls first.
      } else {
        // Move the nulls from the start of the list to the end of the list.
        const auto numNulls = countLeadingNulls(values);
        for (auto i = 0; i < values.size() - numNulls; ++i) {
          values[i] = values[i + numNulls];
        }

        for (auto i = values.size() - numNulls; i < values.size(); ++i) {
          values[i].reset();
        }
      }
    } else {
      if (nullsFirst) {
        // Keeps nulls at the start of the list. Reverse the order of remaining
        // non-null elements.
        const auto numNulls = countLeadingNulls(values);
        std::reverse(values.begin() + numNulls, values.end());
      } else {
        // Reverse the order of all elements.
        std::reverse(values.begin(), values.end());
      }
    }
  }
```

Then, it can be called like this in all 3 places.

```
changeSortingOrder(expectedOrder, ascending, nullsFirst);
```